docs/blog/_posts/2019-09-23-19th-dotty-milestone-release.md

```diff
@@ -0,0 +1,209 @@
+
+All of the experimental syntax related to givens – such as `delegate for`, `given as`, infix-style `given` – is now dropped.
+
+## Colons dropped from class or object definitions
+Now you can define an object as follows:
+
+```scala
+object Bar
+  val x = 10
+
+@main def Test = println(Bar.x)  // 10
+```
+
+In `0.18.1-RC1`, you would have needed to put a colon after `Bar`. The colon was also dropped for traits, classes and enums.
+
+## Allow `given` in pattern bindings
+Consider you have the following monadic flow:
+
+```scala
+for
+  x <- bar
+  res <- foo(given x)
+yield res
+```
+
+Writing entire programs in a monadic flow is not uncommon in functional programming. When working in this style, a situation may arise as shown above: one statement of the monadic flow implicitly depends on the result of another one. It was impossible to declare a pattern variable as a given, which necessitated passing it around explicitly. Not anymore! Now, you can write the above code as follows:
+
+```scala
+for
+  given x: Int <- bar  // Int, or whatever type you are extracting
+  res <- foo
+yield res
+```
+
+Note that the type of the given variable must be specified explicitly.
+
+Full example:
+
+```scala
+def foo(given x: Int): Option[Int] = Some(x * x)
+def bar = Some(10)
+
+@main def Test =
+  for
+    given x: Int <- bar
+    res <- foo
+  yield println(res)
+```
+
+This syntax is allowed anywhere where a pattern is allowed. So you can write:
+
+```scala
+user match
+  case User(_, Some(given email: Email)) => sendEmail
+```
+
+Full example:
+
+```scala
+opaque type Email = String
+object Email
+  def apply(value: String): Email = value
+
+def sendEmail(given m: Email): Unit =
+  println(s"Sent an email to $m")
+
+case class User(name: String, email: Option[Email])
+
+@main def Test =
+  val user = User("Tom", Some(Email("tom@gmail.com")))
+  user match
+    case User(_, Some(given email: Email)) => sendEmail
+```
+
+## Replace given matches by a library method
+Given matches was a feature that allowed to query the implicit scope and execute different logic based on what was found there. We have replaced this feature with a library method called `summonFrom`. You can use it as follows:
+
+```scala
+import compiletime.summonFrom
+
+given Int = 10
+
+@main inline def Test = summonFrom {
+  case str: String => println(s"String $str")
+  case int: Int => println(s"Int $int")  // Int 10
+}
+```
+
+The above code will print "Int 10" since an integer with the value 10 was present in the implicit scope but no String was present.
+
+Notice that we had to define the `Test` method as `inline` since `summonFrom` can only be used from an inline method.
+
+## Wildcard types written with `?`
+You can now use both `_` and `?` to express wildcard types. For example:
+
+```scala
+@main def Test =
+  val xs: List[Int] = (1 to 10).toList
+  xs match
+    case xss: List[?] => println(s"It is a list")
```

Comment:
Add:

This is the first step in a multi-step process to disallow `_` as wildcards so that we can underscores for both terms and type parameters instead. This will make the language more regular.